### DISCUSSION TITLE
JSON data in BelongsToSelect Field

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#2725] [Help] [answered]
I'd like to populate a SelectField with JSON data imported from a Rest API with GuzzleHttp

How can i do ?

**Comment by danharrin** [comment, marked as answer]
You can pass the results into `->options()`

**Reply by eaziweb95** [reply]
ok but i must instantiate the data list before. 
In the MVC Laravel version, i put this instanciation in the constructor

    public function __construct(CommuneRepository $communes)
    {
        $this->communeRepository = $communes;
        parent::__construct();
    }

but it dosen't work in the resource file ...

**Reply by danharrin** [reply]
```php
->options(function (CommuneRepository $communes): array {
    // return array from $communes
})
```

**Reply by eaziweb95** [reply]
Thanks it works !
